.github/workflows/ci.yml

```diff
@@ -40,6 +40,7 @@ jobs:
     steps:
       - uses: actions/checkout@v3
         with:
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
What this would be if we trigger CI manually?

Reply:
```
${{ github.event.pull_request.head.sha || github.sha}}
```
This may more accurate.
- https://github.com/apache/arrow-datafusion/blob/main/.github/workflows/dev.yml#L22